addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -70,17 +70,40 @@ weSnippetEditor.Class.include({
                 size: 'medium',
                 title: _t("Google Map API Key"),
                 buttons: [
-                    {text: _t("Save"), classes: 'btn-primary', close: true, click: async () => {
-                        const newAPIKey = dialog.$('#api_key_input').val() || false;
-                        await this._rpc({
-                            model: 'website',
-                            method: 'write',
-                            args: [
-                                [websiteId],
-                                {google_maps_api_key: newAPIKey},
-                            ],
-                        });
-                        invalidated = true;
+                    {text: _t("Save"), classes: 'btn-primary', click: async (ev) => {
+                        const $apiKeyInput = dialog.$('#api_key_input');
+                        const valueAPIKey = $apiKeyInput.val();
+                        const $apiKeyHelp = dialog.$('#api_key_help');
+                        if (!valueAPIKey) {
+                            $apiKeyInput.addClass('is-invalid');
+                            $apiKeyHelp.text(_t('Enter an API Key'));
+                            return;
+                        }
+                        const $button = $(ev.currentTarget);
+                        $button.prop('disabled', true);
+                        try {
+                            const response = await fetch(`https://maps.googleapis.com/maps/api/staticmap?center=belgium&size=10x10&key=${valueAPIKey}`);
+                            if (response.status === 200) {
+                                await this._rpc({
+                                    model: 'website',
+                                    method: 'write',
+                                    args: [
+                                        [websiteId],
+                                        {google_maps_api_key: valueAPIKey},
+                                    ],
+                                });
+                                invalidated = true;
+                                dialog.close();
+                            } else {
+                                const text = await response.text();
+                                $apiKeyInput.addClass('is-invalid');
+                                $apiKeyHelp.text(_t('Invalid API Key, google returned the follow error: ') + text);
```

Comment:
Typo
```suggestion
                                $apiKeyHelp.text(_t('Invalid API Key, Google returned the following error: ') + text);
```

But actually since you need to change it, `Invalid API Key. The following error was returned by Google:` seems better, no?
Also wrap the Google error in italic at least (or anything to separate it visually from our message)